x-pack/plugins/ml/public/application/components/data_grid/feature_importance/use_classification_path_data.tsx

```diff
@@ -165,9 +199,87 @@ export const buildClassificationDecisionPathData = ({
     }
     return undefined;
   });
+
+  // get the baseline for the current class from the trained_models metadata
+  const baselineClass = baselines.find(
+    (bl) => getStringBasedClassName(bl.class_name) === getStringBasedClassName(currentClass)
+  );
   const filteredFeatureImportance = mappedFeatureImportance.filter(
     (f) => f !== undefined
   ) as ExtendedFeatureImportance[];
 
-  return buildDecisionPathData(filteredFeatureImportance);
+  const finalResult: DecisionPathPlotData = filteredFeatureImportance
+    // sort by absolute importance so it goes from bottom (baseline) to top
+    .sort(
+      (a: ExtendedFeatureImportance, b: ExtendedFeatureImportance) =>
+        b.absImportance! - a.absImportance!
+    )
+    .map((d) => [d[FEATURE_NAME] as string, d[FEATURE_IMPORTANCE] as number, NaN]);
+
+  /**
+   * For binary classification
+   */
+  if (baselines.length === 2) {
+    // transform the numbers into the probability space
+    // starting with the baseline retrieved from trained_models metadata
+    let logOddSoFar = baselineClass?.baseline ? baselineClass?.baseline : 0;
+    for (let i = featureImportance.length - 1; i >= 0; i--) {
+      logOddSoFar += finalResult[i][1];
+      const predictionProbabilitySoFar = Math.exp(logOddSoFar) / (Math.exp(logOddSoFar) + 1);
+      finalResult[i][2] = predictionProbabilitySoFar;
+    }
+    return finalResult;
+  }
+
+  /**
+   * For multiclass classification
+   */
+
+  // first calculate the denominator
+  // (\sum_{x\in{A,B,C}} exp(baseline(x) + \sum_{i=0}^j feature_importance_i(x)))
+  let denominator = 0;
+  for (let x = 0; x < baselines.length; x++) {
```

Comment:
I think it's more common to use `i` 🙂

Reply:
I used `x` here it to match with the `sum_{x\in{A,B,C}}` in the commented formula (since `i` is used to denote featureImportance). I think for clarity, I'll keep the `x` here to refer to class, and `i` to refer to the feature importance.